src/ray/core_worker/lib/java/org_ray_runtime_RayNativeRuntime.cc

```diff
@@ -37,42 +37,46 @@ JNIEXPORT jlong JNICALL Java_org_ray_runtime_RayNativeRuntime_nativeInitCoreWork
   auto job_id = JavaByteArrayToId<ray::JobID>(env, jobId);
   auto gcs_client_options = ToGcsClientOptions(env, gcsClientOptions);
 
-  auto executor_func = [](const ray::RayFunction &ray_function,
-                          const std::vector<std::shared_ptr<ray::RayObject>> &args,
-                          int num_returns,
-                          std::vector<std::shared_ptr<ray::RayObject>> *results) {
-    JNIEnv *env = local_env;
-    RAY_CHECK(env);
-    RAY_CHECK(local_java_task_executor);
-    // convert RayFunction
-    jobject ray_function_array_list =
-        NativeStringVectorToJavaStringList(env, ray_function.GetFunctionDescriptor());
-    // convert args
-    // TODO (kfstorm): Avoid copying binary data from Java to C++
-    jobject args_array_list = NativeVectorToJavaList<std::shared_ptr<ray::RayObject>>(
-        env, args, NativeRayObjectToJavaNativeRayObject);
+  auto task_execution_callback =
+      [](ray::TaskType task_type, const ray::RayFunction &ray_function,
+         const ActorID &actor_id,
+         const std::unordered_map<std::string, double> &required_resources,
+         const std::vector<std::shared_ptr<ray::RayObject>> &args,
+         const std::vector<ObjectID> &arg_reference_ids,
+         const std::vector<ObjectID> &return_ids,
+         std::vector<std::shared_ptr<ray::RayObject>> *results) {
```

Comment:
these added params seems are not used for java. Are these for python specifically?